packages/next/server/next-server.ts

```diff
@@ -706,7 +706,11 @@ export default class NextNodeServer extends BaseServer {
           },
         }
       } catch (err) {
-        if (isError(err) && err.code !== 'ENOENT') throw err
+        // we should only not throw if the error is a ENOENT
+        // when attempting to find the page path
+        if (isError(err) && !(err.code === 'ENOENT' && 'page' in err)) {
```

Comment:
Should we check `err.page === pagePath`  too?

Reply:
I think we should be safe checking the error type which was added in https://github.com/vercel/next.js/pull/37486/commits/db631f40bd647205d3a79a11f27c5f4b49208b76